tests/Agency/StoreTestAPI.cpp

```diff
@@ -0,0 +1,1347 @@
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Ignacio Rodriguez
+/// @author Max Neunhoeffer
+/// @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include "gtest/gtest.h"
+
+#include "Agency/Store.h"
+#include "Mocks/Servers.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Compare.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+namespace arangodb {
+namespace tests {
+namespace store_test_api {
+
+class StoreTestAPI : public ::testing::Test {
+ public:
+  StoreTestAPI() : _server(), _store(_server.server(), nullptr) {
+  }
+ protected:  
+  arangodb::tests::mocks::MockCoordinator _server;
+  arangodb::consensus::Store _store;
+
+  auto readAndCheck(std::string const &json) 
+  {
+    auto q {VPackParser::fromJson(json)};
+    auto result {std::make_shared<VPackBuilder>()};
+    _store.read(q, result);
+    return result;
+  }
+
+  auto write(std::string const &json)
+  {
+    auto q {VPackParser::fromJson(json)};
+    return _store.applyTransactions(q);
+  }
+
+  void writeAndCheck(std::string const &json) 
+  {
+    auto r {write(json)};
+    ASSERT_TRUE(std::all_of(r.begin(), r.end(), [](auto const &result){ return result == consensus::apply_ret_t::APPLIED; }));
+  }
+
+  void assertEqual(std::shared_ptr<velocypack::Builder> result, std::string const &expected_result) const {
+    auto expected {VPackParser::fromJson(expected_result)};
+    ASSERT_TRUE(velocypack::NormalizedCompare::equals(result->slice(), expected->slice()));
+  }
+
+
+};
+
+TEST_F(StoreTestAPI, our_first_test) {
+  std::shared_ptr<VPackBuilder> q
+    = VPackParser::fromJson(R"=(
+        [[{"/": {"op":"delete"}}]]
+      )=");
+  std::vector<consensus::apply_ret_t> v = _store.applyTransactions(q);
+  ASSERT_EQ(1, v.size());
+  ASSERT_EQ(consensus::APPLIED, v[0]);
+  
+  q = VPackParser::fromJson(R"=(
+        ["/x"]
+      )=");
+  VPackBuilder result;
+  ASSERT_TRUE(_store.read(q->slice(), result));
+  VPackSlice res{result.slice()};
+#if 0
+  ASSERT_TRUE(res.isArray() && res.length() == 1);
+  res = res[0];
+#endif
+  ASSERT_TRUE(res.isObject() && res.length() == 0);
+  auto j = VPackParser::fromJson(R"=(
+       {}
+     )=");
+  ASSERT_TRUE(velocypack::NormalizedCompare::equals(j->slice(), result.slice()));
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Test transact interface
+////////////////////////////////////////////////////////////////////////////////
+/*
+TEST_F(StoreTestAPI, testPoll) {
```

Comment:
This test is not needed. It only tests the log poll API, which is not present in the Store.

Reply:
Yes, after our conversation there will be this and another one that I will remove. Thank you.